responsibleai/responsibleai/_managers/causal_manager.py

```diff
@@ -240,6 +240,8 @@ def add(
                 treatment_features[i], treatment_cost[i],
                 alpha, max_tree_depth, min_tree_leaf_samples)
             result.policies.append(policy)
+
+        result._validate_schema()
```

Comment:
It seems in causal it's done during add and load, while for counterfactual/error analysis it's now done for both compute and load?  It seems a bit inconsistent.  I also wonder if we need to move to a similar object to 'CausalResult' for error analysis and counterfactual?

Reply:
I don't know when/why was the implementation for causal changed from add/compute to just compute. I think Chris changed it for some reason. I don't think it is trivial to change to add()/compute() as azureml depends on this. 

>> I also wonder if we need to move to a similar object to 'CausalResult' for error analysis and counterfactual?
For causal, there was no object based interface that was provided by econml. Counterfactuals have a self-explanatiory interface like explanations so I don't think it makes a whole lot of sense to migrate Counterfactuals to that interface. For ErrorAnalysis, we can discuss this if we need to move to CausalResult based object.

Reply:
We had been thinking about removing the `compute()` call, since it's a little pointless when you have live compute. Syncing the removal with the AzureML code would obviously be required, but it could be made a noop in the meantime.

Reply:
We need to decide on the name of the API that does both add and compute. Removing **compute()** is outside of the scope of this PR.